x-pack/plugins/security_solution/common/detection_engine/schemas/request/add_prepackged_rules_schema.test.ts

```diff
@@ -0,0 +1,1395 @@
+    const payload: Partial<AddPrepackagedRulesSchema> = {
+      rule_id: 'rule-1',
+      description: 'some description',
+      from: 'now-5m',
+      to: 'now',
+      index: ['index-1'],
+      name: 'some-name',
+      severity: 'low',
+      interval: '5m',
+      type: 'query',
+      query: 'some query',
+      language: 'kuery',
+      risk_score: 50,
+    };
+
+    const decoded = addPrepackagedRulesSchema.decode(payload);
+    const checked = exactCheck(payload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual([
+      'Invalid value "undefined" supplied to "version"',
+    ]);
+    expect(message.schema).toEqual({});
+  });
+
+  test('[rule_id, description, from, to, index, name, severity, interval, type, query, language, version] does validate', () => {
+    const payload: Partial<AddPrepackagedRulesSchema> = {
+      rule_id: 'rule-1',
+      description: 'some description',
+      from: 'now-5m',
+      to: 'now',
+      index: ['index-1'],
+      name: 'some-name',
+      severity: 'low',
+      interval: '5m',
+      type: 'query',
+      query: 'some query',
+      language: 'kuery',
+      risk_score: 50,
+      version: 1,
+    };
+
+    const decoded = addPrepackagedRulesSchema.decode(payload);
+    const checked = exactCheck(payload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual([]);
+    const expected: AddPrepackagedRulesSchemaDecoded = {
+      rule_id: 'rule-1',
+      risk_score: 50,
+      description: 'some description',
+      from: 'now-5m',
+      to: 'now',
+      index: ['index-1'],
+      name: 'some-name',
+      severity: 'low',
+      interval: '5m',
+      type: 'query',
+      query: 'some query',
+      language: 'kuery',
+      references: [],
+      actions: [],
+      enabled: false,
+      false_positives: [],
+      max_signals: DEFAULT_MAX_SIGNALS,
+      tags: [],
+      threat: [],
+      throttle: null,
+      version: 1,
+      exceptions_list: [],
+    };
+    expect(message.schema).toEqual(expected);
+  });
+
+  test('[rule_id, description, from, to, index, name, severity, interval, type, query, language, risk_score, output_index] does not validate', () => {
+    const payload: Partial<AddPrepackagedRulesSchema> & { output_index: string } = {
+      rule_id: 'rule-1',
+      output_index: '.siem-signals',
+      risk_score: 50,
+      description: 'some description',
+      from: 'now-5m',
+      to: 'now',
+      index: ['index-1'],
+      name: 'some-name',
+      severity: 'low',
+      interval: '5m',
+      type: 'query',
+      query: 'some query',
+      language: 'kuery',
+    };
+
+    const decoded = addPrepackagedRulesSchema.decode(payload);
+    const checked = exactCheck<AddPrepackagedRulesSchema>(
+      (payload as unknown) as AddPrepackagedRulesSchema,
```

Comment:
This is due to `exactCheck` requiring that `original` is constrained to `T`, right? If it were changed to `unknown`:

```ts
export const exactCheck = <T>(
  original: unknown,
  decoded: Either<t.Errors, T>
): Either<t.Errors, T> => {
```

we wouldn't need to perform these casts. But perhaps I'm missing something?

Reply:
I think you're right, I changed it locally and it looks good. I have changed all of these everywhere and I also was able to remove a lot of the templating in a lot of spots too.

Thank you! 🙏

Reply:
nit: I think `findDifferencesRecursive` should also change to receive `unknown`, but if you wanna merge this tonight we can follow up after.